website/javascript/templates/UserProfile.vue

```diff
@@ -814,18 +812,23 @@
               })
             })).then((data) => {
               challenges.forEach((c, i) => {
+                let flag = false
                 c.players.forEach((p) => {
-                  // no bots
-                  if (this.participants[p.user_id].length == 0){
-                    challenges[i].status = 'Paused'
-                  }
-                  // bots compilation is failed
-                  else if (this.participants[p.user_id] &&
-                    this.participants[p.user_id][0].compilation_status !== 'Successful' &&
-                    !c.finished){
-                    challenges[i].status = 'Paused'
-                  } else {
-                    challenges[i].status = c.finished ? 'Completed' : 'In Progress'
+                  if (!flag){
+                    // no bots
+                    if (this.participants[p.user_id].length == 0){
+                      challenges[i].status = 'Paused'
+                      flag = true
+                    }
+                    // bots compilation is failed
+                    else if (this.participants[p.user_id] &&
+                      this.participants[p.user_id][0].compilation_status !== 'Successful' &&
+                      !c.finished){
+                        challenges[i].status = 'Paused'
+                        flag = true
+                      } else {
+                        challenges[i].status = c.finished ? 'Completed' : 'In Progress'
+                      }
```

Comment:
It seems like the check for challenged finished should be before the checks to see if the challenge is paused. In other words. once finished the challenge should always be 'Completed' no matter what the state of the participants bots is in at that point.